ultravox/inference/utils.py

```diff
@@ -10,7 +10,12 @@ def default_device():
 
 
 def default_dtype():
-    return torch.bfloat16 if torch.cuda.is_available() else torch.float32
+    # MPS got bfloat16 support in macOS Sonoma 14.
+    return (
+        torch.bfloat16
```

Comment:
Just double checking: your example shows `float16` is faster. Did you also try it with `bfloat16`?

Reply:
Added a bfloat16 run. There's a lot of variance depending on whether it's the first run or not, but the qualitative assessment is that float16/bfloat16 are about the same and float32 is much slower.

Reply:
why not use `float16` as the default when not using `bfloat16`

Reply:
I assumed that it will generally be slower to do float16 on CPU since the CPU's native type is float32?

Reply:
Hmm, I tried this and float16 is much faster than float32, ~2 TPS vs 0.33 TPS. Changed accordingly.

Reply:
I just tested. Inference in float16 is also significantly faster than in float32